test/Validate.CFG.cpp

```diff
@@ -0,0 +1,601 @@
+using std::stringstream;
+using std::vector;
+
+using ::testing::HasSubstr;
+
+using libspirv::BasicBlock;
+using libspirv::ValidationState_t;
+
+using ValidateCFG = spvtest::ValidateBase<bool>;
+using spvtest::ScopedContext;
+
+namespace {
+
+string nameOps() { return ""; }
+
+template <typename... Args>
+string nameOps(pair<string, string> head, Args... names) {
+  return "OpName %" + head.first + " \"" + head.second + "\"\n" +
+         nameOps(names...);
+}
+
+template <typename... Args>
+string nameOps(string head, Args... names) {
+  return "OpName %" + head + " \"" + head + "\"\n" + nameOps(names...);
+}
+
+class Block {
+  string label_;
+  string body_;
+  SpvOp type_;
+  vector<Block> successors_;
+
+ public:
+  Block(string label, SpvOp type = SpvOpBranch)
+      : label_(label), body_(), type_(type), successors_() {}
+
+  Block &setBody(std::string body) {
+    body_ = body;
+    return *this;
+  }
+
+  operator string() {
+    stringstream out;
+    out << std::setw(8) << "%" + label_ + "  = OpLabel \n";
+    if (!body_.empty()) {
+      out << body_;
+    }
+
+    switch (type_) {
+      case SpvOpBranchConditional:
+        out << "OpBranchConditional %cond ";
+        for (Block &b : successors_) {
+          out << "%" + b.label_ + " ";
+        }
+        break;
+      case SpvOpSwitch: {
+        out << "OpSwitch %one %" + successors_.front().label_;
+        stringstream ss;
+        for (size_t i = 1; i < successors_.size(); i++) {
+          ss << " " << i << " %" << successors_[i].label_;
+        }
+        out << ss.str();
+      } break;
+      case SpvOpReturn:
+        out << "OpReturn\n";
+        break;
+      case SpvOpUnreachable:
+        out << "OpUnreachable\n";
+        break;
+      case SpvOpBranch:
+        out << "OpBranch %" + successors_.front().label_;
+        break;
+      default:
+        assert(1 != 1 && "Unhandled");
+    }
+    out << "\n";
+
+    return out.str();
+  }
+  friend Block &operator>>(Block &curr, vector<Block> successors);
+  friend Block &operator>>(Block &lhs, Block &successor);
+};
+
+Block &operator>>(Block &lhs, vector<Block> successors) {
+  if (lhs.type_ == SpvOpBranchConditional) {
+    assert(successors.size() == 2);
+  } else if (lhs.type_ == SpvOpSwitch) {
+    assert(successors.size() > 1);
+  }
+  lhs.successors_ = successors;
+  return lhs;
+}
+
+Block &operator>>(Block &lhs, Block &successor) {
+  assert(lhs.type_ == SpvOpBranch);
+  lhs.successors_.push_back(successor);
+  return lhs;
+}
+
+string header =
+    "OpCapability Shader\n"
+    "OpMemoryModel Logical GLSL450\n";
+
+string types_consts =
+    nameOps("voidt", "boolt", "intt", "one", "two", "ptrt", "funct") +
+    "%voidt   = OpTypeVoid\n"
+    "%boolt   = OpTypeBool\n"
+    "%intt    = OpTypeInt 32 1\n"
+    "%one     = OpConstant %intt 1\n"
+    "%two     = OpConstant %intt 2\n"
+    "%ptrt    = OpTypePointer Function %intt\n"
+    "%funct   = OpTypeFunction %voidt\n";
+
+TEST_F(ValidateCFG, Default) {
```

Comment:
Not really sure what `Default` as a test-case name refers to.  Is there a more descriptive name?